certora/specs/GeneralAdapter1Reverts.spec

```
@@ -173,20 +98,22 @@ rule revertOrStateChanged(env e, method f, calldataarg args) filtered {
          f.selector != sig:onMorphoFlashLoan(uint256, bytes).selector &&
          // Property checked in a different way.
          f.selector != sig:nativeTransfer(address, uint256).selector &&
-         f.selector != sig:wrapNative(uint256, address).selector &&
          f.selector != sig:unwrapNative(uint256, address).selector &&
          f.selector != sig:erc20Transfer(address, address, uint256).selector &&
-         f.selector != sig:erc20TransferFrom(address, address, uint256).selector &&
         // Property doesn't hold for the following.
+         f.selector != sig:erc20TransferFrom(address, address, uint256).selector &&
          f.selector != sig:morphoFlashLoan(address, uint256, bytes).selector &&
          f.selector != sig:permit2TransferFrom(address, address, uint256).selector
 }{
     // Safe require as the initiator can't be zero when executing a bundle.
     require Bundler.initiator() != 0;
+    // Safe require as the initiator can't be the adatper.
+    require Bundler.initiator() != currentContract;
+
 
     storage storageBefore = lastStorage;
 
     f@withrevert(e, args);
 
-    assert !lastReverted => storageChanged(storageBefore, lastStorage);
+    assert !lastReverted => storageBefore != lastStorage;
```

```suggestion
    assert !lastReverted => (storageBefore != lastStorage) || (nativeBalancesBefore != nativeBalanceAfter);
```
could we do this and merge everything into one rule?

---

No because `lastStorage` includes `nativeBalances`, this is not the problem. We need extra hypothesis for the rules to go through.

---

This is also why I don't like this style, just sending ETH to GeneralAdapter1 satisfies the rule.

---

but you need to send it all the time. And the generalAdapter doesn't enforce that you send ETH. I still think that the rule is useful like that